### DISCUSSION TITLE
Macedonian voice for tts

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#160] [Archived]
<USER>>>> Ivona_Milanova<br/> <DATE> [March 13, 2020, 6:44am]<br/> <br/> Hi, *slash* <br/> I am creating a TTS for Macedonian Language. I have collected 9 hours of<br/> data and I am training a Tacotron model.<br/> <br/> This is the alignment I get after 128000 steps. However I get noise at<br/> the end of the train file. Is this caused because of silences at the end<br/> of audio files?<br/> <br/> 178<br/> <br/> This is the configuration I am using:<br/> <br/> 'audio':{ *slash* <br/> // Audio processing parameters *slash* <br/> 'num_mels': 80, // size of the mel spec frame. *slash* <br/> 'num_freq': 1025, // number of stft frequency levels. Size of the linear<br/> spectogram frame. *slash* <br/> 'sample_rate': 22050, // DATASET-RELATED: wav sample-rate. If different<br/> than the original data, it is resampled. *slash* <br/> 'frame_length_ms': 50, // stft window length in ms. *slash* <br/> 'frame_shift_ms': 12.5, // stft window hop-lengh in ms. *slash* <br/> 'preemphasis': 0.98, // pre-emphasis to reduce spec noise and make it<br/> more structured. If 0.0, no -pre-emphasis. *slash* <br/> 'min_level_db': -100, // normalization range *slash* <br/> 'ref_level_db': 20, // reference level db, theoretically 20db is the<br/> sound of air. *slash* <br/> 'power': 1.5, // value to sharpen wav signals after GL algorithm. *slash* <br/> 'griffin_lim_iters': 60,// [ *slash* #griffin-lim] iterations. 30-60<br/> is a good range. Larger the value, slower the generation. *slash* <br/> // Normalization parameters *slash* <br/> 'signal_norm': true, // normalize the spec values in range *slash* [0, 1 *slash* ] *slash* <br/> 'symmetric_norm': true, // move normalization to range *slash* [-1, 1 *slash* ] *slash* <br/> 'max_norm': 4, // scale normalization to range *slash* [-max_norm, max_norm *slash* ]<br/> or *slash* [0, max_norm *slash* ] *slash* <br/> 'clip_norm': true, // clip normalized values into the range. *slash* <br/> 'mel_fmin': 0.0, // minimum freq level for mel-spec. *slash* ~50 for male and<br/> *slash* ~95 for female voices. Tune for dataset!! *slash* <br/> 'mel_fmax': 8000.0, // maximum freq level for mel-spec. Tune for<br/> dataset!! *slash* <br/> 'do_trim_silence': false // enable trimming of slience of audio as you<br/> load it. LJspeech (false), TWEB (false), Nancy (true) *slash* <br/> },<br/> <br/> 'distributed':{<br/> 'backend': 'nccl',<br/> 'url': 'tcp: *slash* / *slash* /localhost:54321'<br/> },<br/> <br/> 'reinit_layers': [],<br/> <br/> 'model': 'Tacotron', // one of the model in models/ <br/> 'grad_clip': 1, // upper limit for gradients for clipping.<br/> 'epochs': 3000, // total number of epochs to train.<br/> 'lr': 0.0001, // Initial learning rate. If Noam decay is active, maximum learning rate.<br/> 'lr_decay': false, // if true, Noam learning rate decaying is applied through training.<br/> 'warmup_steps': 4000, // Noam decay steps to increase the learning rate from 0 to 'lr'<br/> 'memory_size': -1, // ONLY TACOTRON - size of the memory queue used fro storing last decoder predictions for auto-regression. If < 0, memory queue is disabled and decoder only uses the last prediction frame.<br/> 'attention_norm': 'sigmoid', // softmax or sigmoid. Suggested to use softmax for Tacotron2 and sigmoid for Tacotron.<br/> 'prenet_type': 'original', // 'original' or 'bn'.<br/> 'prenet_dropout': true, // enable/disable dropout at prenet.<br/> 'windowing': false, // Enables attention windowing. Used only in eval mode.<br/> 'use_forward_attn': false, // if it uses forward attention. In general, it aligns faster.<br/> 'forward_attn_mask': false,<br/> 'transition_agent': false, // enable/disable transition agent of forward attention.<br/> 'location_attn': true, // enable_disable location sensitive attention. It is enabled for TACOTRON by default.<br/> 'loss_masking': true, // enable / disable loss masking against the sequence padding.<br/> 'enable_eos_bos_chars': false, // enable/disable beginning of sentence and end of sentence chars.<br/> 'stopnet': true, // Train stopnet predicting the end of synthesis.<br/> 'separate_stopnet': true, // Train stopnet seperately if 'stopnet==true'. It prevents stopnet loss to influence the rest of the model. It causes a better model, but it trains SLOWER.<br/> 'tb_model_param_stats': false, // true, plots param stats per layer on tensorboard. Might be memory consuming, but good for debugging.<br/> <br/> 'batch_size': 32, // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.<br/> 'eval_batch_size':16, <br/> 'r': 7, // Number of decoder frames to predict per iteration. Set the initial values if gradual training is enabled. <br/> 'gradual_training': [[0, 7, 32], [1, 5, 32], [50000, 3, 32], [130000, 2, 16], [290000, 1, 16]], // ONLY TACOTRON - set gradual training steps [first_step, r, batch_size]. If it is null, gradual training is disabled. <br/> 'wd': 0.000001, // Weight decay weight<br/> 'checkpoint': true, // If true, it saves checkpoints per 'save_step'<br/> 'save_step': 1000, // Number of training steps expected to save traninpg stats and checkpoints.<br/> 'print_step': 25, // Number of steps to log traning on console.<br/> 'batch_group_size': 0, //Number of batches to shuffle after bucketing.<br/> <br/> 'run_eval': true,<br/> 'test_delay_epochs': 5, //Until attention is aligned, testing only wastes computation time.<br/> 'test_sentences_file': null, // set a file to load sentences to be used for testing. If it is null then we use default english sentences.<br/> 'min_seq_len': 9, // DATASET-RELATED: minimum text length to use in training<br/> 'max_seq_len': 182, // DATASET-RELATED: maximum text length<br/> 'output_path': '../keep/', // DATASET-RELATED: output path for all training outputs.<br/> 'num_loader_workers': 2, // number of training data loader processes. Don't set it too big. 4-8 are good values.<br/> 'num_val_loader_workers': 2, // number of evaluation data loader processes.<br/> 'phoneme_cache_path': 'mozilla_mk_phonemes', // phoneme computation is slow, therefore, it caches results in the given folder.<br/> 'use_phonemes': false, // use phonemes instead of raw characters. It is suggested for better pronounciation.<br/> 'phoneme_language': 'mk', // depending on your target language, pick one from https://github.com/bootphon/phonemizer#languages<br/> 'text_cleaner': 'basic_cleaners',<br/> 'use_speaker_embedding': false, // use speaker embedding to enable multi-speaker learning.<br/> 'style_wav_for_test': null, // path to style wav file to be used in TacotronGST inference.<br/> 'use_gst': false, // TACOTRON ONLY: use global style tokens<br/> <br/> 'datasets': // List of datasets. They all merged and they get different speaker_ids.<br/> [<br/> {<br/> 'name': '',<br/> 'path': '',<br/> 'meta_file_train': 'metadata_train.csv',<br/> 'meta_file_val': 'metadata_val.csv'<br/> }<br/> ]<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/macedonian-voice-for-tts]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [March 13, 2020, 8:35am]<br/> <br/> > 'do_trim_silence': false<br/> <br/> Try with setting 'do_trim_silence: true'<br/> <br/> While you at it you might also want to adapt mel_fmin and mel_fmax<br/> setting, e.g. 'mel_fmin': 50.0<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> sanjaesc<br/> <DATE> [March 13, 2020, 9:20am]<br/> <br/> Do you get the same noise when generating sentences? If not then you<br/> should be fine.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [March 13, 2020, 10:14am]<br/> <br/> That noise you will only be probably get during training time, not<br/> inference.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [March 15, 2020, 2:45pm]<br/> <br/> noise in training time is normal due to padding<br/> <br/> [Archived Post]<br/>
